src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,170 @@
 package core.basesyntax;
 
+//import jdk.internal.util.ArraysSupport;
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
+
 public class ArrayList<T> implements List<T> {
+
+
+    private static final int def_capacity = 10;
+    private static final Object[] empty_arraydata = {};
+
+
+    private T[] array = (T[]) new Object[def_capacity];
+    private int size;
+    private int cursor;
+
+
+    private String errorMsg(int index) {
+        return "Index: " + index + ", Size: " + size;
+    }
+
+
+    private Object[] grow() {
+        int oldCapacity = array.length;
+        double multiplicator = 1;
+        if (oldCapacity == size) {
+            multiplicator = 1.5;
+        }
+        if (oldCapacity > 0 || array != empty_arraydata) {
+            int newCapacity = (int) (oldCapacity * multiplicator + 1);
+            return array = Arrays.copyOf(array, newCapacity);
+        } else {
+            return array = (T[]) new Object[def_capacity];
+        }
+    }
```

Comment:
1) Make it void and use your updated 'array' variable
2) Pass the 'newSize" parameter
3) You need the only one check here - if (newSize > array.length)
4) Create a new constant field for multiplicator
```suggestion
    private void grow(int newSize) {
        if (newSize > array.length){
            //here you need to create a new 'array with new size' using the created constant 'MULTIPLICATOR' and call 'System.arraycopy()' to copy your current 'array' to the newly created 'array with new size' and then assign 'array with new size' to your 'array' variable
        }
    }
```